Feat/Link sampler selection lock to TC API Type

leandrojofre opened the pull request:
## Changes
As requested, sampler selection lock is no longer tied to preset samplers, it is now linked to Text Completion API Type.
A lot of changes had to be made in `samplerSelect.js`, the logic of using `power_user.selectSamplers`, `userShownSamplers`, `userDisabledSamplers`, `selectsampler` as JQuery data, and `selectedSamplers` (localforage) was constantly conflicting. `power_user.selectSamplers` and the other two (that were also `power_user`) were replaced with directly using the data stored in `selectsampler` for each sampler block. It is virtually the same, as `power_user.selectSamplers` was never really saved into user settings, but reset every time you change API Type or load ST.

## Commits
- [Update](https://github.com/SillyTavern/SillyTavern/commit/9c70ab2d5ebb4b549c22a1072ff2c7abf1654a88) - Revert connection profile change TC commands load order
> This change was made to prevent preset tied samplers from breaking, such thing doesn't exist anymore.
- [Update](https://github.com/SillyTavern/SillyTavern/commit/884611ab490e2e4472992816106e82ea2d68080f) - Unlink preset renaming logic from sampler selection lock
- [Update](https://github.com/SillyTavern/SillyTavern/commit/c733081e49dd67146c16d33d2aab3161c670a78b) - Link TC sampler selection lock to API Type
- [Fix](https://github.com/SillyTavern/SillyTavern/commit/fa14fd7232d2ce4714d5c91f0f0243065e8b84a7) - Clean comments
- [Update](https://github.com/SillyTavern/SillyTavern/commit/7c1c1eb99ef1593b128eae8e3786599abd3b41a3) - Use localforage selectedSamplers and data selectsampler as main sampler selection storage
- [Fix](https://github.com/SillyTavern/SillyTavern/commit/b7f6b17d0c6df4ab10bf4b198fa279614afe26f5) - ESLint errors

## Checklist:

- [X] I have read the [Contribution guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

leandrojofre committed:
Update - Revert connection profile change TC commands load order

This change was made to prevent preset tied samplers from breaking, such thing doesn't exist anymore.

leandrojofre committed:
Update - Unlink preset renaming logic from sampler selection lock

leandrojofre committed:
Update - Link TC sampler selection lock to API Type

leandrojofre committed:
Fix - Clean comments

leandrojofre committed:
Update - Use localforage selectedSamplers and data selectsampler as main sampler selection storage

leandrojofre committed:
Fix - ESLint errors

leandrojofre committed:
Update - Change lock tooltip with correct description

Cohee1207 commented:
@codex review

Cohee1207 reviewed (approved):
Yeah that works for me. Gave it a quick prod, appears fine.

leandrojofre committed:
Fix - Move and await sampler select localforage initializer to finish

Tried to move it to the most reasonable place possible that allowed to make an await.

leandrojofre commented on public/scripts/textgen-settings.js:
```diff
@@ -779,7 +777,7 @@ async function getStatusTextgen() {
 export function initTextGenSettings() {
-    loadPresetSelectedSamplers();
+    loadApiSelectedSamplers();
 
```
Aight, fixed.
[Fix](https://github.com/SillyTavern/SillyTavern/pull/4782/commits/550c49aff1ee563a9ab835a6c2a8adaf829d3779) - Move and await sampler select localforage initializer to finish `> Tried to move it to the most reasonable place possible that allowed to make an await.`

Cohee1207 replied:
I think it'd be fine to make loadTextGenSettings async instead. It's only ever used in one specific place (in settings first load init)

leandrojofre replied:
Change made, here: [Fix](https://github.com/SillyTavern/SillyTavern/pull/4782/commits/16f6dfa6fe9a79ed2069c1f5afbd31d158dca689) - Make loadTextGenSettings async to load API selected samplers

Cohee1207 replied:
Thanks.

leandrojofre committed:
Fix - Make loadTextGenSettings async to load API selected samplers
